tests/test_auto_tick.py

```diff
@@ -1,14 +1,147 @@
+import copy
+import logging
+import tempfile
+from pathlib import Path
 from unittest import mock
-from unittest.mock import ANY, MagicMock
+from unittest.mock import ANY, MagicMock, create_autospec
 
 import pytest
 
-from conda_forge_tick.auto_tick import run_with_tmpdir
+from conda_forge_tick.auto_tick import _prepare_feedstock_repository, run_with_tmpdir
 from conda_forge_tick.contexts import ClonedFeedstockContext, FeedstockContext
+from conda_forge_tick.git_utils import (
+    DryRunBackend,
+    GitPlatformBackend,
+    RepositoryNotFoundError,
+)
 
 demo_attrs = {"conda-forge.yml": {"provider": {"default_branch": "main"}}}
 
 
+def test_prepare_feedstock_repository_success():
+    backend = create_autospec(GitPlatformBackend)
+
+    with tempfile.TemporaryDirectory() as tmpdir_str:
+        tmpdir = Path(tmpdir_str)
+        cloned_context = ClonedFeedstockContext(
+            feedstock_name="pytest",
+            attrs=demo_attrs,
+            default_branch="main",
+            local_clone_dir=tmpdir,
+        )
+
+        assert (
+            _prepare_feedstock_repository(
+                backend, cloned_context, "new_branch", "base_branch"
+            )
+            is True
+        )
+
+        backend.fork.assert_called_once_with("conda-forge", "pytest-feedstock")
+
+        backend.clone_fork_and_branch.assert_called_once_with(
+            upstream_owner="conda-forge",
+            repo_name="pytest-feedstock",
+            target_dir=tmpdir,
+            new_branch="new_branch",
+            base_branch="base_branch",
+        )
+
+
+def test_prepare_feedstock_repository_repository_not_found(caplog, fake_lazy_json):
+    backend = create_autospec(GitPlatformBackend)
+
+    caplog.set_level(logging.WARNING)
+
+    with tempfile.TemporaryDirectory() as tmpdir_str:
+        tmpdir = Path(tmpdir_str)
+
+        demo_attrs_copy = copy.deepcopy(demo_attrs)
+
+        attrs = copy.deepcopy(fake_lazy_json)
+        pr_info = copy.deepcopy(fake_lazy_json)
```

Comment:
Having to deepcopy this fixture is odd. Can we import the class directly and just make new ones as needed?

Reply:
fixed in 7f67c49